scripts/publish-metadata.py

```diff
@@ -0,0 +1,120 @@
+import argparse
+import sys
+import requests
+import json
+import subprocess
+
+DESCRIPTION = """
+Uploads a given block of metadata to the Terra Data Catalog
+
+To run this script you must authed as an "admin" on the
+ specified environment's catalog resource
+
+"""
+
+
+def generate_access_token(user):
+    auth_as_user(user)
+
+    printProc = subprocess.Popen(
+        ["gcloud auth print-access-token"], stdout=subprocess.PIPE, shell=True
+    )
+    (out, err) = printProc.communicate()
+
+    return out.decode("ASCII").strip()
+
+
+def auth_as_user(user):
+    proc = subprocess.Popen(
+        [f"gcloud auth login {user} --brief"], stdout=subprocess.PIPE, shell=True
+    )
+    proc.communicate()
+
+
+def create_dataset(env, accessToken, storage_system, storage_system_id, entry):
+    headers = {
+        "Authorization": f"Bearer {accessToken}",
+        "Content-Type": "application/json",
+    }
+
+    request = {
+        "storageSystem": str(storage_system),
+        "storageSourceId": str(storage_system_id),
+        "catalogEntry": json.dumps(entry),
+    }
+    print(headers)
+    response = requests.post(
```

Comment:
If the dataset already exists, `post` will fail and you need to use `put` instead. Or use `get` before `post` to see if it already exists. Maybe something like https://github.com/DataBiosphere/terra-data-catalog/blob/main/scripts/catalog-tests/generate-tdr-snapshot-metadata.py#L68

Not sure if it's time to think about avoiding code duplication between this script and `generate-tdr-snapshot-metadata.py` and `ingest-hca-collection.py`. At some point we'll probably want to refactor these scripts.

Reply:
And/or we could look at using the python client codegen from swagger for this code.

Reply:
TOL: 
So, this is specifically intended to just handle `post`. I didn't want to handle updating too, because I didn't know if that did smart updates. However, it looks like its always a full replacement? So I can change this to be `publish-or-update` metadata.

My other concern is that in order to `get`, we would need the data-catalog id, which we don't necessarily have at this point. So we would need to `list`, `filter` by `storageSourceId`, and then `put`. This is doable. It is a bit more work, but that seems worth it to me.

Reply:
> However, it looks like its always a full replacement?

Yes, that's right.

> My other concern is that in order to get, we would need the data-catalog id, which we don't necessarily have at this point. So we would need to list, filter by storageSourceId, and then put. This is doable. It is a bit more work, but that seems worth it to me.

Yeah my text didn't match the code I linked to, which does the `list` and filters the result.

Reply:
We can't do that. We don't return `storageSourceId` with our `list` endpoint. So we would need to add that information (as well as storage system) to our `list` response before we could detect if it exists already/figure out which dataset is the one we care about.

Reply:
Hmm… looking at the code you linked actually, looks like we do have `dct:identifier` for TDR entries, but not for workspace entries yet. I'll add that to the list. However, either way, we are missing the storage system in both cases, so we'd need to add that to our response.

Reply:
We can't retry after error on POST, because we have no way to tell what to PUT. You need to get the data catalog ID and there's no way to derive that information from the storage source id and storage system. We could potentially add that information, but that would be a service change to add more information to our return objects.

Reply:
It looks like HTTP status 409 (conflict) can be used here. If our POST API receives data which would result in a duplicate entry, we can return a 409 status response with the URL to the existing resource (possibly in the `Content-Location:` header). See https://www.rfc-editor.org/rfc/rfc9110.html#name-409-conflict.

Reply:
I hesitate to block this on updating our endpoint to return a better error message though. I can file a ticket to add the url to the 409 error that you get back when a dataset for a particular storage id has already been created.

Reply:
I agree, let's not block this PR on this feature. We can go back and add support once we add the support in the service code. In the meantime we're not yet running this in bulk so I think it's OK to require manual intervention in the case of duplicate entries.